Mage.Sets/src/mage/cards/e/ElspethResplendent.java

```diff
@@ -144,8 +144,9 @@ public boolean apply(Game game, Ability source) {
                 permanent.addCounters(CounterType.SHIELD.createInstance(), source, game);
             }
         }
+        cards.remove(card);
```

Comment:
this line wasn't necessary, the `retainZone` function will handle the card no longer being in library and also handles a situation like Grafdigger's Cage preventing the card from entering the battlefield

Reply:
Does the version of "putCardsOnBottomOfLibrary" that takes a single card not care where the card is coming from?